src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -374,17 +452,12 @@ public override GroupBy GroupBy(int columnIndex, DataFrame parent)
             }
         }
 
-        public void ApplyElementwise(Func<T, T> func)
+        public void ApplyElementwise(Func<T?, long, T?> func)
         {
-            foreach (ReadOnlyDataFrameBuffer<T> buffer in _columnContainer.Buffers)
+            Parallel.For(0, Length, index =>
```

Comment:
Why use a `Parallel.For` here? Have you done benchmarks that show this actually makes a performance improvement?

Reply:
Benchmarks will follow soon. Also, based on the benchmarks I just ran in a private branch, I'm going to revert this change. `Parallel.For` is about 50% slower than the regular for loop on my machine. It was on a large 50000 element column too, so not something I expected. I'll revert the change.

Reply:
Oh, the `Parallel.For` loop is 50% faster than a regular for loop, but 50% slower than the unrolled version on the left. The unrolled version drills down to the Span first and then starts applying `func`, whereas the regular for loop would involve indexing calls for each loop iteration in addition to applying `func`. I bet cache effects are in play here. I wonder if the generated IL code makes it obvious, but that investigation is for later this week.

Reply:
That is what I expected, I've seen this in the past. If you are doing a relatively small amount of work inside the loop, using a `Parallel.For` will slow your program down because of the overhead due to switching and coordinating between threads.